### PR TITLE
KAFKA-6382: Make ProducerConfig and ConsumerConfig constructors public

### PR DESCRIPTION
**Pull request by mjsax**

### CODE REVIEW

**Comment by mjsax** [comment]
DO NOT MERGE
This is the PR for KIP-243 that is not voted yet.

Related to #4315 -- if the other PR get's merge first, we should update this PR to make use of the available constructors. If this PR gets merged first, #4315 need to be updated before merged.

Call for review @bbejeck @guozhangwang @dguy

File: clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java [hunk 1]
```diff
@@ -598,10 +598,18 @@ public KafkaConsumer(Map<String, Object> configs) {
...
      *            won't be called in the consumer when the deserializer is passed in directly.
      */
-    public KafkaConsumer(Map<String, Object> configs,
+    public KafkaConsumer(final Map<String, Object> configs,
                          Deserializer<K> keyDeserializer,
                          Deserializer<V> valueDeserializer) {
-        this(new ConsumerConfig(ConsumerConfig.addDeserializerToConfig(configs, keyDeserializer, valueDeserializer)),
+        this(new ConsumerConfig(
+            ConsumerConfig.addDeserializerToConfig(
+                new Properties() {
```
Comment (bbejeck) [inline clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java]: I understand the need for the DBI here on `Properties` and below, but I'm wondering if we could add a static method to `org.apache.kafka.common.utils.CollectionUtils` that takes a map and returns `Properties` instead? (unless that would require an update to the KIP)

**Comment by bbejeck** [comment]
one minor comment, otherwise LGTM.

File: clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java (hunk 1 above)
Comment (mjsax) [reply clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java]: Seems to be public API... :(

**Comment by mjsax** [comment]
Retest this please

**Comment by mjsax** [comment]
Updated according to KIP discussion. Also use ProducerConfig to access BATCH_SIZE default value and get rid or hard coded value.

Call for review @bbejeck @guozhangwang @dguy @hachikuji

File: clients/src/main/java/org/apache/kafka/clients/consumer/ConsumerConfig.java [hunk 2]
```diff
@@ -476,8 +475,11 @@ public static Properties addDeserializerToConfig(Properties properties,
             newProperties.put(VALUE_DESERIALIZER_CLASS_CONFIG, valueDeserializer.getClass().getName());
         return newProperties;
     }
+    public ConsumerConfig(Properties props) {
```
Comment (guozhangwang) [inline clients/src/main/java/org/apache/kafka/clients/consumer/ConsumerConfig.java]: nit: newline.

**Review by guozhangwang — APPROVED** [review]
nit comment, otherwise lgtm.

**Review by bbejeck — APPROVED** [review]
thanks @mjsax, updates LGTM

**Comment by guozhangwang** [comment]
@mjsax Please feel free to merge after addressing the nit comment. LGTM.

**Pull request merged**